python/lsst/obs/decam/instrument.py

```diff
@@ -0,0 +1,86 @@
+# This file is part of obs_decam.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Butler instrument description for the Dark Energy Camera.
+"""
+
+__all__ = ("DarkEnergyCamera",)
+
+import os
+
+from lsst.afw.cameraGeom import makeCameraFromPath, CameraConfig
+from lsst.daf.butler.instrument import Instrument
+from lsst.obs.decam.decamFilters import DECAM_FILTER_DEFINITIONS
+
+from lsst.utils import getPackageDir
+
+
+class DarkEnergyCamera(Instrument):
+    """Gen3 Butler specialization class for the DECam instrument.
+    """
+    filterDefinitions = DECAM_FILTER_DEFINITIONS
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        packageDir = getPackageDir("obs_decam")
+        self.configPaths = [os.path.join(packageDir, "config")]
+
+    @classmethod
+    def getName(cls):
+        return "DECam"
+
+    def getCamera(self):
+        path = os.path.join(getPackageDir("obs_decam"), "decam", "camGeom")
+        config = CameraConfig()
+        config.load(os.path.join(path, "camera.py"))
+        return makeCameraFromPath(
+            cameraConfig=config,
+            ampInfoPath=path,
+            shortNameFunc=lambda name: name.replace(" ", "_"),
+        )
+
+    def register(self, registry):
+        camera = self.getCamera()
+        dataId = {"instrument": self.getName()}
+        obsMax = 2**31
+        registry.addDimensionEntry("instrument", dataId,
+                                   entries={"detector_max": 64,
+                                            "visit_max": obsMax,
+                                            "exposure_max": obsMax})
+
+        for detector in camera:
+            registry.addDimensionEntry(
+                "detector", dataId,
+                detector=detector.getId(),
+                name=detector.getName(),
+                # DECam doesn't have rafts
+                raft=None
```

Comment:
It does have rafts. There are 2: N and S. In fact there are some support scripts in obs_decam that try to reconstruct the raft and ObservationInfo will report the raft. (which is called "detector_group").

Reply:
@ctslater do you have a better word than raft? We have to balance something generic (like ObservationInfo.detector_group) vs something that LSST people will like.

Reply:
Discussing this offline with @ctslater : the things on DECam and HSC are not rafts, they are groups of detectors. I think we can continue to get by not referring to them at all (I've removed the `raft=None` and comment). If we do decide we want a way to group them in the future, we could call them `detector_group`s here, but we would also want the `raft` to be an alias for that for LSST.

For this ticket, lets just leave it with no such groups.

Reply:
This needs to be fixed in the gen 3 data model and can't be fixed on this ticket. The evidence is clear that there are scripts in obs_decam or obs_subaru  that do try to work out detector groupings from the detector name and we should be up front about this possibility.